rumqttc/src/websockets.rs

```diff
@@ -8,6 +10,33 @@ pub enum UrlError {
     Parse(#[from] http::uri::InvalidUri),
 }
 
+#[derive(Debug, thiserror::Error)]
+pub enum ValidationError {
+    #[error("Websocket response does not contain subprotocol header")]
+    SubprotocolHeaderMissing,
+    #[error("Websocket subprotocol header: {0}")]
+    SubprotocolMqtt(String),
+}
+
+pub(crate) fn validate_response_headers(
+    response: Response<Option<Vec<u8>>>,
+) -> Result<(), ValidationError> {
+    let val = response
+        .headers()
+        .get("Sec-WebSocket-Protocol")
+        .ok_or(ValidationError::SubprotocolHeaderMissing)?;
+
+    let sub_protocol = val
+        .to_str()
+        .map_err(|_| ValidationError::SubprotocolHeaderMissing)?;
+
+    if sub_protocol != "mqtt" {
+        return Err(ValidationError::SubprotocolMqtt(sub_protocol.to_owned()));
+    }
```

Comment:
as per specs [ref](https://http.dev/ws#sec-websocket-protocol):
> This header can appear multiple times within the request, or instead, multiple subprotocols can be listed in a comma-delimited format
> 

here, we aren't covering cases where multiple protocols are listed. e.g.
`Sec-WebSocket-Protocol: something-idk, mqtt`

suggestions: we can split the header at `,` and then check if `mqtt` exists!